File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int n=s.length();
        int a=-1,b=-1,c=-1,substrings=0;
        for(int i=0;i<n;i++)
        {
            if(s[i]=='a') a=i;
            else if(s[i]=='b') b=i;
            else c=i;
            if(a!=-1 && b!=-1 && c!=-1)
            {
                substrings+=1+min({a,b,c});
            }
        }
        return substrings;
        
        // //vector<int> result(3,-1);
        // int substrings=0;
        // for(int i=0;i<n;i++)
        // {
        //     //int ind=s[i]-'a';
        //     //result[ind]=i;
        //     // if(result[0]!=-1 && result[1]!=-1 && result[2]!=-1)
        //     // {
        //     //     substrings+=1+min({result[0],result[1],result[2]});
        //     // }
        // }
        // return substrings;
    }
};

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int n = s.length();
        int a = -1, b = -1, c = -1;
        int count = 0;

        for(int i=0;i<n;i++) {
            char ch = s.charAt(i);
            if(ch == 'a') a = i;
            else if(ch == 'b') b = i;
            else c = i;

            if(a != -1 && b != -1 && c != -1) {
                count += 1 + Math.min(a,Math.min(b,c));
            }
        }
        return count;
    }
}